core/market/decentralized/src/market.rs

```diff
@@ -158,6 +160,22 @@ impl MarketService {
         // TODO: shouldn't remove precede negotiation unsubscribe?
         Ok(self.matcher.unsubscribe_demand(demand_id, id).await?)
     }
+
+    pub async fn get_proposal(
+        &self,
+        proposal_id: &ProposalId,
+        _id: &Identity,
+    ) -> Result<Proposal, MarketError> {
+        // TODO: Autorization
```

Comment:
What are the dangers of unauthorized access to proposal?

Reply:
Negotiations are private, meaning that only negotiating parties should know content of Proposals (Other identities on the same daemon probably shouldn't have access either).
But we scheduled thinking about Identities to Milestone 2, so we may decide, that we don't need authorization here.